### ISSUE TITLE
Wrong error message when calling static method using dot notation

### ISSUE THREAD

**Issue by defyrlt** [rust-lang/rust#30391]
[Playpen](http://is.gd/iisPSu). For some reasons I wasn't able to get this error message when struct is in the same crate.

``` rust
extern crate rustc;

use rustc::middle::ty::MethodCall;

fn main() {
    let mc = MethodCall.expr(); // ERROR: unresolved name `MethodCall` [E0425]
    // let mc = MethodCall; // same error here
}
```

Here's the error message I get when struct is in the same crate

``` rust
<anon>:10:13: 10:16 error: `Foo` is the name of a struct or struct variant, but this expression uses it like a function name [E0423]
```

I think it should be raised in the case with crate too.

**Comment by birkenfeld** [comment]
This appears to be fixed (in nightly at least):
```
z.rs:6:14: 6:24 error: `MethodCall` is the name of a struct or struct variant, but this expression uses it like a function name [E0423]
z.rs:6     let mc = MethodCall.expr(); // ERROR: unresolved name `MethodCall` [E0425]
                    ^~~~~~~~~~
z.rs:6:14: 6:24 help: run `rustc --explain E0423` to see a detailed explanation
z.rs:6:14: 6:24 help: did you mean to write: `MethodCall { /* fields */ }`?
```

**Comment by Mark-Simulacrum** [comment]
I believe the error message today indicates the problem fairly well, though it could show the `Foo::foo` form, so not closing.

```rust
struct Foo;

impl Foo {
    fn foo() {
    }
}

fn bar() {
    Foo.foo();
}
```

```
error[E0599]: no method named `foo` found for type `Foo` in the current scope
 --> test.rs:9:9
  |
9 |     Foo.foo();
  |         ^^^
  |
  = note: found the following associated functions; to be used as methods, functions must have a `self` parameter
note: candidate #1 is defined in an impl for the type `Foo`
 --> test.rs:4:5
  |
4 | /     fn foo() {
5 | |     }
  | |_____^

error: aborting due to previous error(s)
```
